proto/src/lib.rs

```diff
@@ -133,4 +136,27 @@ pub mod health {
     }
 }
 
+#[cfg(all(feature = "protobuf-codec", not(feature = "prost-codec")))]
```

Comment:
Why does these ones have a more complicated `cfg` block compared to the above? Is it related to defaults?

Reply:
It is because Cargo can arbitrarily combine features so we must be able to support both features at the same time. If we use the simpler cfg here, then we would get two versions of the same name in the module.